src/server/view/DetectionPanel.java

```diff
@@ -151,7 +157,303 @@ public DetectionPanel() {
         PerformanceMetricsSpinner.setBounds(161, 229, 55, 25);
         this.add(PerformanceMetricsSpinner);
         
+        upperfaceSpinner.addChangeListener(new ChangeListener() {
+            @Override
+            public void stateChanged(ChangeEvent e) {
+                float upperfaceValue = (float)((double) upperfaceSpinner.getValue());
+                String upperfaceExp = (String)UpperfaceComboBox.getSelectedItem();
+                if(upperfaceExp=="Raise Brow")
+                {
+                    ServerDataSingleton.getInstance().getFaceData().getExpressiveData().setRaiseBrow(upperfaceValue);
+                    ServerDataSingleton.getInstance().getFaceData().getExpressiveData().setFurrowBrow(0);
+                }
+                if(upperfaceExp=="Furrow Brow")
+                {
+                    ServerDataSingleton.getInstance().getFaceData().getExpressiveData().setFurrowBrow(upperfaceValue);
+                    ServerDataSingleton.getInstance().getFaceData().getExpressiveData().setRaiseBrow(0);
+                }
+
+            }
+        });
+         
+        UpperfaceComboBox.addActionListener(
```

Comment:
We need to implement action listener interface in the class.

And use e.getSource() to get the value changed. and not have anonymous implementation